c7n/resources/glue.py

```diff
@@ -80,3 +80,50 @@ def delete_connection(self, r):
     def process(self, resources):
         with self.executor_factory(max_workers=2) as w:
             list(w.map(self.delete_connection, resources))
+
+
+@resources.register('glue-dev-endpoint')
+class GlueDevEndpoint(QueryResourceManager):
+
+    class resource_type(object):
+        service = 'glue'
+        enum_spec = ('get_dev_endpoints', 'DevEndpoints', None)
+        detail_spec = None
+        id = name = 'EndpointName'
+        date = 'CreatedTimestamp'
+        dimension = None
+        filter_name = None
+
+    permissions = ('glue:GetDevEndpoints',)
+
+
+@GlueDevEndpoint.action_registry.register('delete')
+class DeleteDevEndpoint(BaseAction):
+    """Deletes public Glue Dev Endpoints
+
+    :example:
+
+    .. code-block: yaml
+
+        policies:
+          - name: delete-public-dev-endpoints
+            resource: glue-dev-endpoint
+            filters:
+              - PublicAddress: present
+            actions:
+              - type: delete
+    """
+    schema = type_schema('delete')
+    permissions = ('glue:DeleteDevEndpoint',)
+
+    def delete_dev_endpoint(self, r):
+        client = local_session(self.manager.session_factory).client('glue')
```

Comment:
@kapilt Can you point me to an example of this as I am not sure what format it needs to be in?

Reply:
there's a few examples using w.submit and passing the client, in this case, its probably easier to just skip the threading , not sure the cardinality counts here would make it worthwhile.

Reply:
So it would be like this example:
```
def process(self, clusters):
        with self.executor_factory(max_workers=2) as w:
            futures = []
            for db_set in chunks(clusters, size=5):
                futures.append(
                    w.submit(self.process_db_set, db_set))
            for f in as_completed(futures):
                if f.exception():
                    self.log.error(
                        "Exception deleting redshift set \n %s",
                        f.exception())
```